### ISSUE TITLE
BUG: HierarchicalCfC per-cycle training re-advances live temporal state and lacks historical-start semantics

### ISSUE THREAD

**Issue by Tristan-Stoltz-ERC** [Luminous-Dynamics/symthaea#3584]
Related: #3574, #3578, #3579, #3580, #3583
Affected area: `TemporalNetwork::train_step_from` / `HierarchicalCfC::train_step` / cognitive-loop training phase

## Summary
The 2026-07-17 sequence-prediction training fix introduced a rolling queue of historical temporal-state snapshots so the per-cycle training pair `(enc_{t-1} -> enc_t)` can start from the evolution state at the end of cycle `t-2`, rather than from the already-advanced live state.

That correction currently applies only to HdcLtc.

For HierarchicalCfC:
- `save_evolution_state()` returns `None`, so the history queue is never populated;
- `TemporalNetwork::train_step_from(start, ...)` falls through to `train_step_bptt(...)` regardless of `start`;
- HierarchicalCfC `train_step_bptt` delegates to `hcfc.train_step(...)`;
- `HierarchicalCfC::train_step(...)` calls `forward_hierarchical(...)`, which advances hidden state, hierarchy `total_steps`, `recent_slow_outputs`, and potentially actual tau state through top-down modulation.

Therefore a learning cycle can advance/mutate the live temporal hierarchy **again after the planning step**, using the training pair rather than preserving the online inference trajectory.

## Current semantic mismatch

The training-phase comment states that fallback trainers for non-HdcLtc backends “at least do not corrupt live state.” That assumption needs correction for HierarchicalCfC.

Conceptually today:

```text
planning:
    live_hcfc.step(enc_t, dt)          // intended online evolution

training gate fires:
    hcfc.train_step(enc_{t-1}, enc_t)  // another forward_hierarchical on LIVE state
                                       // + parameter updates

cycle ends with replay/training-evolved state,
not just the planned online state
```

This conflates two distinct effects:
- intended checkpoint/parameter learning;
- unintended mutation of live inference evolution state.

## Historical-start problem
The intended temporally correct training start is the state at end of `t-2` for pair `(enc_{t-1} -> enc_t)`.

Because HierarchicalCfC cannot currently snapshot/restore:
- it cannot reproduce that start;
- training uses whatever live state exists after current-cycle planning/prediction activity;
- the training loss/gradient is therefore conditioned on the wrong temporal context;
- the training forward pass changes live context again.

## Why this matters
This can distort:
- temporal memory/context across cycles;
- learned prediction dynamics;
- actual tau adaptation state;
- prediction error and coherence;
- experiments comparing HierarchicalCfC with single-scale CfC/HdcLtc;
- any claim that the cognitive loop's live temporal state reflects one online step per cycle.

It also means historical HCFC runs with learning enabled may combine #3580 prediction resets, #3583 consolidation state destruction, and this per-cycle training mutation.

## Immediate containment
Until #3578 exact snapshot/restore and a qualified historical-start training profile exist, do not silently claim temporally correct/non-corrupting per-cycle HierarchicalCfC training.

Conservative options:
- disable/suspend HCFC online per-cycle training under the affected profile with explicit telemetry;
- train a detached exact copy/checkpoint and apply only qualified parameter deltas if that can be done without importing its evolution state;
- after #3578, snapshot live state, restore the correct historical training start on a training execution context, update intended parameters, then restore live inference evolution state under the resulting checkpoint according to an explicit profile.

Do not use projected `read_state()/inject()` as historical restore.

## Parameter/evolution separation theorem
A qualified solution must define:

```text
online inference evolution state
!= training-start evolution state
!= learned/checkpoint parameter state
```

Training may update parameters while the online inference trajectory remains at the state produced by the online planning path, unless the model explicitly preregisters learning itself as a semantic evolution event.

If parameter changes make restoring the pre-training inference state under new weights non-equivalent, that interaction must be defined and tested rather than ignored.

## Training profiles
At minimum distinguish:

```text
FrozenInference                 // no parameter update
LiveStateTrainingLegacy         // historical behavior, explicitly labeled
HistoricalStartDetachedTraining // desired sequence-correct profile
OnlineAdaptiveSemanticTraining  // only if training mutation is intentionally part of state evolution
```

Do not call all of these simply `train_step` in evidence receipts.

## Required tests
- demonstrate pre-fix HCFC live snapshot/state changes solely because the training phase ran after planning;
- demonstrate hierarchy `total_steps`/context/tau changes from training forward work where applicable;
- training-disabled twin preserves one-step-per-cycle online trajectory;
- once exact snapshots exist, historical-start profile uses the committed end-of-(t-2) state;
- parameter updates can be separated from online evolution-state preservation under the chosen qualified profile;
- `TrainingMethod::Bptt`, `BpttWithSpsaFallback`, and SPSA/fallback behavior are each audited for HCFC state mutation;
- missing historical snapshot cannot silently fall back while the receipt still claims sequence-correct training;
- HdcLtc historical-start behavior does not regress;
- classic CfC behavior is separately characterized rather than assumed equivalent.

## Evidence impact
HCFC cognitive-loop evidence generated with learning enabled should record this as a known temporal-training confound until rerun under an explicit qualified profile.

This does not establish that the learned weights are useless or that HierarchicalCfC is inferior; it establishes that current online-state and training-state semantics are entangled.

## Non-goals
- no claim that learning should always be detached from live state;
- no top-down tau algorithm redesign;
- no architecture ranking;
- no consciousness conclusion;
- no production backend switch.
